preview_test.go

```diff
@@ -700,6 +690,55 @@ func Test_Extract(t *testing.T) {
 	}
 }
 
+func TestPresetValidation(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		name         string
+		dir          string
+		input        preview.Input
+		presetAssert map[string]assertPreset
+	}{
+		{
+			name:  "preset failure",
```

Comment:
Should we maybe add some extra test cases here?
* Valid preset with prebuilds (happy path)
* Multiple missing required parameters
* Invalid parameter value (not missing, but wrong). Example: Value must be between 1 and 10
* Presets with `instances = 0` or no prebuilds block are skipped